diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ee455c..1a3c297 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,12 +12,16 @@ endif()
 
 # Set KissFFT source and header files.
 set(SOURCES "kiss_fft.c"
-            "tools/kiss_fftr.c")
+            "tools/kiss_fftr.c"
+            "tools/kiss_fftnd.c"
+            "tools/kiss_fftndr.c")
 
 set(HEADERS "kissfft.hh"
             "kiss_fft.h"
             "_kiss_fft_guts.h"
-            "tools/kiss_fftr.h")
+            "tools/kiss_fftr.h"
+            "tools/kiss_fftnd.h"
+            "tools/kiss_fftndr.h")
 
 # Create KissFFT static library.
 add_library(${PROJECT_NAME} STATIC ${SOURCES} ${HEADERS})
